cmd/kpod/ps.go

```diff
@@ -0,0 +1,350 @@
+			ctrID = ctr.ID
+			runningFor = units.HumanDuration(time.Since(ctr.State.Created))
+			command = getCommand(ctr.ImageCreatedBy)
+			imageName = ctr.FromImage
+			mounts = getMounts(ctr.Mounts, opts.noTrunc)
+			ports = getPorts(ctr.Config.ExposedPorts)
+			size = units.HumanSize(float64(ctr.SizeRootFs))
+			labels = getLabels(ctr.Labels)
+
+			switch ctr.State.Status {
+			case "stopped":
+				status = "Exited (" + strconv.FormatInt(int64(ctr.State.ExitCode), 10) + ") " + runningFor + " ago"
+			case "running":
+				status = "Up " + runningFor + " ago"
+			default:
+				status = "Created"
+			}
+
+			if !opts.noTrunc {
+				ctrID = ctr.ID[:idTruncLength]
+				imageName = getImageName(ctr.FromImage)
+			}
+
+			params := psOutputParams{
+				ID:         ctrID,
+				Image:      imageName,
+				Command:    command,
+				CreatedAt:  ctr.State.Created,
+				RunningFor: time.Since(ctr.State.Created),
+				Status:     status,
+				Ports:      ports,
+				Size:       size,
+				Names:      ctr.Name,
+				Labels:     labels,
+				Label:      "",
+				Mounts:     mounts,
+			}
+			output = append(output, params)
+		}
+	}
+	var out formats.Writer
+
+	switch opts.format {
+	case "":
+		out = outStruct{output: output, opts: opts}
+	case JSONstr:
+		out = formats.JSONstruct{Output: toGenericPS(output)}
+	default:
+		out = formats.StdoutTemplate{Output: toGenericPS(output), Template: opts.format}
+	}
+
+	formats.Writer(out).Out()
+	return nil
+}
+
+// getCommand gets the actual command from the whole command
+func getCommand(cmd string) string {
+	start := strings.Index(cmd, "[")
+	if start == -1 {
+		return ""
+	}
+	start += len("[")
+	end := strings.Index(cmd, "]")
+	return cmd[start:end]
+}
+
+// getImageName shortens the image name
+func getImageName(img string) string {
+	arr := strings.Split(img, "/")
+	if arr[0] == "docker.io" && arr[1] == "library" {
+		img = strings.Join(arr[2:], "/")
+	} else if arr[0] == "docker.io" {
+		img = strings.Join(arr[1:], "/")
+	}
+	return img
+}
+
+// getLabels converts the labels to a string of the form "key=value, key2=value2"
+func getLabels(labels fields.Set) string {
+	var arr []string
+	if len(labels) > 0 {
+		for key, val := range labels {
+			temp := key + "=" + val
+			arr = append(arr, temp)
+		}
+		return strings.Join(arr, ",")
+	}
+	return ""
+}
+
+// getLabelVal gets the value of the key given for the label
+func getLabelVal(key string, labels fields.Set) string {
+	if val, ok := labels[key]; ok {
+		return val
+	}
+	return ""
+}
+
+// getMounts converts the volumes mounted to a string of the form "mount1, mount2"
+// it truncates it if noTrunc is false
+func getMounts(mounts []specs.Mount, noTrunc bool) string {
+	var arr []string
+	if len(mounts) == 0 {
+		return ""
+	}
+	for _, mount := range mounts {
+		if noTrunc {
+			arr = append(arr, mount.Source)
+			continue
+		}
+		tempArr := strings.SplitAfter(mount.Source, "/")
+		if len(tempArr) >= 3 {
+			arr = append(arr, strings.Join(tempArr[:3], ""))
```

Comment:
`strings.splitAfter` keeps the character you are splitting "after", so I can join it without "/".